### PR TITLE
added a min_eos_p parameter, defined 'SupressTokensLogitsProcessor' c…

### PR DESCRIPTION
**Pull request by anishpsoni**
…lass and created a 'LogitsProcessorList' containing both a custom logits processor and a 'MinLengthLogitsProcessor'.

# What does this PR do?
1. It adds a custom logits processor (CustomLogitsProcessor) to the generate method.
2. Threshold for EOS Probability:
3. In this case the LPL list, it includes the original suppress_tokens_logits_processor, the custom custom_logits_processor, and a MinLengthLogitsProcessor to ensure that the generated output has a minimum length of 1 token.

Fixes # (issue)
#26672

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by LysandreJik** [comment]
cc @ylacombe

**Comment by ylacombe** [comment]
Hi @anishsoni29, thanks for your help here! however I think @isaac-chung already started working on it yesterday, as you can see [here](https://github.com/huggingface/transformers/issues/26672#issuecomment-1752105521). He even already proposed PR #26675.

It's really nice to see your keen motivation here, though! What would you think of contributing to other issues? We have a [docstrings sprint](https://github.com/huggingface/transformers/issues/26638) running this month, if you need any ideas !

**Comment by ylacombe** [comment]
#26675 merged!

**Comment by Cazforshort** [comment]
So how do I set min_eos_p in the actual code?
